### PR TITLE
fix: (2.37) Fetch events based on user organisation units [DHIS2-11755]

### PR DESCRIPTION
**Pull request by zubaira**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java [hunk 1]
```diff
@@ -832,27 +834,59 @@ public ImportSummaries deleteEvents( List<String> uids, boolean clearSession )
...
+                {
+                    organisationUnits
+                        .addAll( organisationUnitService.getOrganisationUnitWithChildren( orgUnit.getUid() ) );
+                }
+                else if ( OrganisationUnitSelectionMode.CHILDREN.equals( orgUnitSelectionMode ) )
+                {
+                    organisationUnits.add( orgUnit );
+                    organisationUnits.addAll( orgUnit.getChildren() );
+                }
+                else // SELECTED
```
Comment (Bekkalizer) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java:859]: What happens if OrganisationUnitSelectionMode is ACCESSIBLE, CAPTURE or ALL?

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java [hunk 2]
```diff
@@ -832,27 +834,59 @@ public ImportSummaries deleteEvents( List<String> uids, boolean clearSession )
...
-                organisationUnits.addAll( orgUnit.getChildren() );
+                Set<OrganisationUnit> children = userOrgUnits.stream()
+                    .map( OrganisationUnit::getChildren )
+                    .flatMap( Collection::stream )
+                    .collect( Collectors.toSet() );
+
+                organisationUnits.addAll( children );
             }
-            else // SELECTED
+            else
```
Comment (Bekkalizer) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java:887]: It seems we need to implement different behavior for OrganisationUnitSelectionMode ACCESSIBLE, CAPTURE and ALL?

Comment (zubaira) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java:887]: https://github.com/dhis2/dhis2-core/pull/8711

**Review by stian-sandvold — APPROVED** [review]

**Comment by zubaira** [comment]
fixed in https://github.com/dhis2/dhis2-core/pull/8711

**Pull request closed without merging**